google/cloud/internal/curl_impl.cc

```diff
@@ -654,6 +654,10 @@ Status CurlImpl::MakeRequest(CurlImpl::HttpMethod method,
   using HttpMethod = CurlImpl::HttpMethod;
   handle_.SetOption(CURLOPT_CUSTOMREQUEST, HttpMethodAsChar(method));
   handle_.SetOption(CURLOPT_UPLOAD, 0L);
+  if (options_.has<CurlFollowLocationOption>()) {
```

Comment:
When the `!has` case is equivalent to `T{}`, and with a `bool` option in particular, I would proffer that it is better (faster and clearer) to drop the `has` check.